File: companion/spss/r/b06.R
veri <- read.csv("companion/spss/csv/b06.csv")
stopifnot(nrow(veri) == 395, ncol(veri) == 13,
          !anyNA(veri))
basit <- subset(veri, srs40 == 1)
tabakali <- subset(veri, strat40 == 1)
stopifnot(nrow(basit) == 40, nrow(tabakali) == 40)
ozet <- function(degerler)
  c(n=length(degerler), mean=mean(degerler),
    sd=sd(degerler))
print(cbind(tam_dosya=ozet(veri$g3),
            basit_rastgele=ozet(basit$g3)))
okul <- factor(tabakali$school, c(1, 2), c("GP", "MS"))
frekans <- table(okul)
print(frekans); print(100 * prop.table(frekans))
print(nrow(tabakali)); print(sum(tabakali$strw))
print(weighted.mean(tabakali$g3, tabakali$strw))

File: companion/spss/python/b06.py
import pandas as pd
import numpy as np

veri = pd.read_csv("companion/spss/csv/b06.csv")
assert veri.shape == (395, 13)
assert not veri.isna().any().any()
basit = veri.loc[veri.srs40 == 1].copy()
tabakali = veri.loc[veri.strat40 == 1].copy()
assert len(basit) == len(tabakali) == 40
print(pd.DataFrame({
    "tam_dosya": veri.g3.agg(["count", "mean", "std"]),
    "basit_rastgele": basit.g3.agg(["count", "mean", "std"])
}))
okul = tabakali.school.map({1: "GP", 2: "MS"})
frekans = okul.value_counts().reindex(["GP", "MS"])
print(frekans); print(100 * frekans / len(tabakali))
print(len(tabakali)); print(tabakali.strw.sum())
print(np.average(tabakali.g3, weights=tabakali.strw))